### PR TITLE
SC: Add TransactionStatements functions to work with lists

### PR DESCRIPTION
**Pull request by bchamagne**
# Description

Add TransactionStatements functions to work with lists.
Authorize Keyword lists in actions. I transformed the whitelist into a blacklist (whitelisting all keyword and blacklisted some keywords from some functions) 

Part of #826

## Type of change

- New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Tests added and tested with the explorer

# Checklist:

- My code follows the style guidelines of this project
- I have performed a self-review of my own code
- I have commented my code, particularly in hard-to-understand areas
- I have made corresponding changes to the documentation
- My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature works
- New and existing unit tests pass locally with my changes
- Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: lib/archethic/contracts/interpreter/transaction_statements.ex [hunk 1]
```diff
@@ -200,6 +200,38 @@ defmodule Archethic.Contracts.Interpreter.TransactionStatements do
...
+  """
+  @spec add_token_transfers(Transaction.t(), list(list())) :: Transaction.t()
+  def add_token_transfers(tx = %Transaction{}, args) when is_list(args) do
+    Enum.reduce(args, tx, &add_token_transfer/2)
+  end
+
+  @doc """
+  Add multiple UCO transfers
+  """
+  @spec add_uco_transfers(Transaction.t(), list(list())) :: Transaction.t()
```
Comment (samuelmanzanera) [inline lib/archethic/contracts/interpreter/transaction_statements.ex:230]: Could you add doctests ?

**Review by samuelmanzanera — APPROVED** [review]

**Comment by bchamagne** [comment]
Latest commit added plenty of unit tests related to another branch (the one with the reduce fn). I suggest not to read the tests in `describe "reduce/3" do`

File: lib/archethic/contracts/interpreter/action.ex [hunk 2]
```diff
@@ -146,140 +146,37 @@ defmodule Archethic.Contracts.ActionInterpreter do
...
   defp prewalk(
-         node = {{:atom, "to"}, address},
-         acc = {:ok, %{scope: {"add_uco_transfer", {:actions, _}}}}
+         node = {{:atom, key}, _args},
+         _acc = {:ok, %{scope: {:function, "add_uco_transfer", {:actions, _}}}}
        )
-       when is_binary(address) do
-    {node, acc}
+       when key != "to" and key != "amount" do
+    throw({:error, "invalid add_uco_transfer arguments", node})
```
Comment (samuelmanzanera) [inline lib/archethic/contracts/interpreter/action.ex:171]: I would prefer whitelisting instead of blacklisting

**Review by samuelmanzanera — APPROVED** [review]
Ok, I think the PR does the work.
I'm not really fan of the blacklisting, but ok for now.
We should later provide a better way with whitelisting a term and blacklisting the remaining, maybe using a dedicated parser for each function.

**Review by Neylix — APPROVED** [review]

**Pull request merged**